blocks/quote-carousel/quote-carousel.js

```diff
@@ -0,0 +1,162 @@
+const DEFAULT_SCROLL_INTERVAL_MS = 6000;
+const numChildren = {};
+const scrollInterval = {};
+const event = new Event('startAutoScroll');
+
+/**
+ * Get key to retrieve data for carousel block
+ *
+ * @param {Element} block
+ * @returns {string}
+ */
+function getDataKey(block) {
+  return block.querySelector('div > div > div:nth-child(2)').innerText;
+}
+
+/**
+ * Stop auto-scroll animation
+ *
+ * @param {Element} block
+ */
+function stopAutoScroll(block) {
+  const key = getDataKey(block);
+  const interval = scrollInterval[key];
+  clearInterval(interval);
+  scrollInterval[key] = undefined;
+}
+
+/**
+ * Return index for active slide
+ *
+ * @param {Element} block
+ * @returns {number}
+ */
+function getCurrentSlideIndex(block) {
+  return [...block.querySelectorAll('.carousel-content .item')].findIndex(
+    (child) => child.getAttribute('active') === 'true',
+  );
+}
+
+/**
+ * Switch between slides
+ *
+ * @param {number} nextIndex
+ * @param {Element} block
+ */
+function switchSlide(nextIndex, block) {
+  const key = getDataKey(block);
+  const slidesContainer = block.querySelector('.carousel-content');
+  const currentIndex = getCurrentSlideIndex(slidesContainer);
+  const prevButton = block.querySelector('.controls-container button[name="prev"]');
+  const nextButton = block.querySelector('.controls-container button[name="next"]');
+  const indexElement = block.querySelector('.controls-container .index');
+  indexElement.textContent = nextIndex + 1;
+  if (currentIndex === 0) {
+    // enable previous button
+    prevButton.removeAttribute('disabled');
+  } else if (nextIndex === 0) {
+    indexElement.dispatchEvent(event);
+    // disable previous button
+    prevButton.setAttribute('disabled', true);
+  } else if (nextIndex === (numChildren[key] - 1)) {
+    // disable next button
+    nextButton.setAttribute('disabled', true);
+    stopAutoScroll(block);
+  } else if (currentIndex === (numChildren[key] - 1)) {
+    // disable next button
+    nextButton.removeAttribute('disabled');
+  }
+  slidesContainer.children[currentIndex].removeAttribute('active');
+  slidesContainer.children[nextIndex].setAttribute('active', true);
+  slidesContainer.style.transform = `translateX(-${nextIndex * 100}%)`;
+}
+
+/**
+ * Start auto scroll
+ *
+ * @param {Element} block
+ * @param {number} interval
+ */
+function startAutoScroll(block, interval = DEFAULT_SCROLL_INTERVAL_MS) {
+  const key = getDataKey(block);
+  scrollInterval[key] = setInterval(() => {
+    const currentIndex = getCurrentSlideIndex(block);
+    switchSlide((currentIndex + 1) % numChildren[key], block);
+  }, interval);
+}
+
+/**
+ * Returns block content from the spreadsheet
+ *
+ * @param {string} type
+ * @returns {Promise<any>}
+ */
+async function getBlockContent(type) {
+  const url = `/drafts/quotes/helix-quotes.json?sheet=${type}`;
```

Comment:
Rather than hard code the URL for the sheet, we can have that be the parameter to the Block, including the tab within the sheet. This way if it needs to move it can.

Reply:
updated